index.js

```diff
@@ -138,47 +179,71 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
+// *** Uncomment the lines below to test
 
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+let num = 0
+
+function takeATicketNumber(line) {
+  num +=1
+  line.push(num)
+  return `Welcome. You are ticket number ${num}`
+}
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
+
++++ "THE CODE BELOW DIDN'T WORK AND I'M A LITTLE SHAKY ON CLOSURES" +++ 
+
+ 
 // console.log("%cScope & Closures - Question 2", "color: red")
 
 // const newLine = []
 // const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+
+// let ticket = 0 
+
+// function ticketNumberGeneratorFunc(){
+//   function takeATicketNumber(line) {
+//     ticket += 1
+//     line.push(ticket)
+//     console.log`Welcome. You are ticket number ${ticket}`
+//   }
+//   return takeATicketNumber
+// }
+
```

Comment:
So close!! Don't worry if you're shaky or not completely understanding this one. This was just an advanced, tricky, bonus one and you did so well, especially for this only being the first day with Javascript! 

The `ticket` variable is currently global. Check out the `Lexical scope` and `Closure` section of this document [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) and see if you realize what you need to change to make this work.